### PR TITLE
Switch to BSD 2-Clause license and alternate licensing options

### PR DESCRIPTION
**Pull request by JeromeMartinez**
While studying [change of MediaConch license](https://mediaarea.net/blog/2018/03/20/Why-we-changed-MediaConch-license), I remarked that some people have a weird interpretation of the word "software" in MIT license, interpreting MIT license as like the zlib license (i.e. without binary redistribution clause) e.g. in [LLVM licensing doc](https://llvm.org/docs/DeveloperPolicy.html#license), and LLVM is backed by Apple so I prefer to explicitly avoid that: I would like that copyright owners are indicated also in binary distributions.

For the [same reasons as for MediaConch](https://mediaarea.net/blog/2018/03/20/Why-we-changed-MediaConch-license) (long story short: avoiding issues with incompatible license requests), I wish to offer alternate licenses possibility, so it is also added in this PR.

@retokromer please review.

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -1,21 +1,26 @@
-MIT License
+BSD 2-Clause License
```
Comment (retokromer) [inline LICENSE]: The official name is `2-Clause BSD`, isn’t it?

Comment (JeromeMartinez) [reply LICENSE]: No. "BSD 2-Clause" when space is acceptable else "BSD-2-Clause".
[default GitHub template](https://github.com/MediaArea/RAWcooked/community/license/new?branch=master&template=bsd-2-clause)
[SPDX entry for BSD 2-Clause](https://spdx.org/licenses/BSD-2-Clause.html)

Comment (retokromer) [reply LICENSE]: Hmm… on their own [website](https://opensource.org/licenses/BSD-2-Clause) it’s some ambiguous: 
bildschirmfoto 2018-03-23 um 18 59 17

Comment (JeromeMartinez) [reply LICENSE]: This is not the BSD website.
[FreeBSD license](https://www.freebsd.org/copyright/freebsd-license.html)

I strongly suggest to keep GitHub template so it is correctly flagged by GitHub, and SPDX (standard for license strings in open source repositories) has the same naming order too.

**Comment by JeromeMartinez** [comment]
> (I’ll fix later HTML5 ;-)

Oops :)

File: LICENSE (hunk 1 above)
Comment (retokromer) [reply LICENSE]: It’s fine with me.

**Comment by retokromer** [comment]
The `README.md` needs to be modified as well.

**Comment by JeromeMartinez** [comment]
`README.md` as well as `.cpp` / `.h` and some Linux distro files have been updated.

**Comment by retokromer** [comment]
- Should we not be added to the `.cpp` and `.h` as well?
- Alternate licences make things a little more complicated, but I am not against them at all.

**Comment by JeromeMartinez** [comment]
> Should we not be added to the .cpp and .h as well?

You talk about alternate licenses? I think file headers should not be too complex, the alternative licenses are intended for rare usages, I don't want to make thing more complex everywhere, and file headers point to the License.html file. It is more on the principle to say that we agree on that, and if a lawyer want more we would provide a specific document.
The primary license is BSD.

**Comment by retokromer** [comment]
>You talk about alternate licenses?

No, about our company. I plenty agree with keeping the file headers simple (only BSD)!

**Comment by JeromeMartinez** [comment]
> No, about our company.

Oops, it was not intended to silently hide something, I reused my classic headers I didn't catch the issue. I did not change the (c) line in this PR.

Additional commit for changing source file headers with both names.

**Pull request merged**